rfc/rfc-0000-allow-resources-to-report-status.md

```diff
@@ -0,0 +1,320 @@
+          - key: status.loadBalancer
+            value: {}
+```
+
+# How it Works
+[how-it-works]: #how-it-works
+
+A stamped resource can be in one of three states:
+1. 'Succeeded'
+2. 'Failed'
+3. 'Unknown'
+
+It is up to each template author to determine what "success" and "failure" mean for the given resource the template is stamping out. "Unknown" strictly represents that Cartographer has not be able to determine success or failure.
+
+A `ClusterResourceStatusRule` requires defining matchers for both the `Succeeded` state and the `Failed` state.
+The matchers in the `Succeeded` state are ANDed together, whereas the matchers in the `Failed` state are ORed. That is, a resource is considered to have `Succeeded` only if _all of_ the matchers are fulfilled, but is considered to have `Failed` if _any one of_ the matchers are fulfilled.
+
+The supported matchers are `matchConditions` and `matchFields`. Each is an array, and an author can define either or both of these matchers for a given state.
+
+For `matchConditions`, authors need only define the `type` and `status` of the condition for Cartographer to read. In the owner status, the full condition will be reflected (with the `reason`, `message`, etc.)
+For `matchFields`, the spec is based off of `matchFields` in blueprint resource options. Authors need to define the `key` and `operator`, and potentially the `values` (if the operator is `In` or `NotIn`). Authors can also optionally define a `messagePath`, which is a path in the resource where Cartographer can pull off a message to display in the owner status.
+
+Each template will now have a new field in the spec `resourceStatusRuleRef` where authors can specify the `ClusterResourceStatusRule` to use for that template. If no `ClusterResourceStatusRule` is defined, Cartographer will default to listing the resource as `Succeeded` once it has been successfully applied to the cluster and any relevant outputs have been read off the resource.
+
+Each owner will display a new condition in the status called `ResourcesSuceeded` which will only be updated to `True` when all resources in the blueprint have succeeded. Cartographer will not update the top level state of an owner to `Ready: True` until `ResourcesSucceeded` is `True`.
+
+# Migration
+[migration]: #migration
+
+Given that Cartographer will default to `Succeeded` once the resource is applied and outputs are read, this will not be a breaking change.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+- The owner status will potentially grow significantly, based on how many matchers are in a `ClusterResourceStatusRule`.
+- We only allow for reflecting three states, though this design allows for creating more states in the future.
+- This design does not allow authors to define the states themselves.
+
+# Alternatives
+[alternatives]: #alternatives
+
+- What other designs have been considered?
+  - The `ClusterResourceStatusRule` could instead be inlined in the templates.
```

Comment:
I think another alternative here is allowing both options:

- inline, if you'd like a one-off that is very specific and unlikely to be shared (e.g., for a configmap)
- ref, if something like `ready: true | ready: false` where it'd be reused a lot

Reply:
I think this is no longer relevant as we have updated the RFC to suggest inlining instead.